From d913d6f1bbb7f0545bdf127567630abcfdc3472a Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:11 -0400
Subject: [infiniband] ocrdma: Fix a sparse warning

Message-id: <1a92ef859cb9a1534446b573a90220ca7465e3e1.1411571481.git.dledford@redhat.com>
Patchwork-id: 96646
O-Subject: [Patch RHEL7 36/48] RDMA/ocrdma: Fix a sparse warning
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Selvin Xavier <selvin.xavier@emulex.com>

Fix the warning about the usage of plain integer as NULL pointer.

Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 1b09a0c29034eaf68bad9f8821531ebd194af3f7)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
index 105659d..a4d27c7 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
@@ -661,7 +661,7 @@ static void ocrdma_dispatch_ibevent(struct ocrdma_dev *dev,
 {
  struct ocrdma_qp *qp = NULL;
  struct ocrdma_cq *cq = NULL;
- struct ib_event ib_evt = { 0 };
+ struct ib_event ib_evt;
  int cq_event = 0;
  int qp_event = 1;
  int srq_event = 0;
@@ -674,6 +674,8 @@ static void ocrdma_dispatch_ibevent(struct ocrdma_dev *dev,
  if (cqe->cqvalid_cqid & OCRDMA_AE_MCQE_CQVALID)
   cq = dev->cq_tbl[cqe->cqvalid_cqid & OCRDMA_AE_MCQE_CQID_MASK];
 
+ memset(&ib_evt, 0, sizeof(ib_evt));
+
  ib_evt.device = &dev->ibdev;
 
  switch (type) {
-- 
1.7.1